src/ecredis.erl

```diff
@@ -0,0 +1,420 @@
+-module(ecredis).
+-behaviour(gen_server).
+
+%% API.
+-export([start_link/1,
+         remap_cluster/2,
+         get_eredis_pid/2,
+         qp/2,
+         q/2]).
+
+%% Callbacks for gen_server.
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+%% Type definition.
+-include_lib("eunit/include/eunit.hrl").
+-include("eredis_cluster.hrl").
+-record(state, {
+    cluster_name :: string(),
+    init_nodes :: [#node{}],
+    slots_cache :: tuple(), %% Tuple with all the Redis slots. Each element indexes into slots_maps.
+    slots_maps :: tuple(), %% Tuple with all the Redis nodes and Redis slots mapped on them.
+    version :: integer()  %% Used to avoid unnecessary refresh of Redis slots.
+}).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% API
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec start_link([]) -> {ok, pid()}.
+start_link({ClusterName, Args}) ->
+    gen_server:start_link({local, ClusterName}, ?MODULE, [ClusterName, Args], []).
+
+
+%% Looks up eredis Pid for Slot.
+-spec get_eredis_pid(ClusterName::atom(), Slot::integer()) -> 
+    {Pid::pid(), Version::integer()} | undefined.
+get_eredis_pid(ClusterName, Slot) ->
+    Result = ets:lookup(ets_table_name(ClusterName), Slot),
+    case Result of
+        [] -> undefined;
+        [{_, Result2}] -> Result2
+    end. 
+
+
+%% Remaps various redis slots.
+-spec remap_cluster(ClusterName::atom(), Version::integer()) -> {Version::integer()}.
+remap_cluster(ClusterName, Version) ->
```

Comment:
callers dont need this api right? because this is already being handled in this query calls?
it's definitely good to have when building out a separate package. 
I was wondering, specifically for our case: do you think we will use this?

Reply:
We could force the cluster to remap instead of waiting for it to happen automatically.